### PR TITLE
added callback for cancel button

### PR DESCRIPTION
**Pull request by mannprerak2**
Added **cancel callbacks** in all 3 widgets. ( #11 )
Also changed null values in ok button to make it behave as expected from Buttons in Flutter

Smoke tests are failing due to some issue with Flare

### CODE REVIEW

File: lib/src/asset.dart [hunk 1]
```diff
@@ -80,7 +84,10 @@ class AssetGiffyDialog extends StatelessWidget {
                           shape: RoundedRectangleBorder(
                               borderRadius:
                                   BorderRadius.circular(buttonRadius)),
-                          onPressed: () => Navigator.of(context).pop(),
+                          onPressed: () {
+                            Navigator.of(context).pop();
+                            onCancelButtonPressed?.call();
```
Comment (xsahil03x) [inline lib/src/asset.dart]: Let the user decide the whole ```onPressed``` body, instead of popping it.

File: lib/src/asset.dart [hunk 2]
```diff
@@ -92,7 +99,8 @@ class AssetGiffyDialog extends StatelessWidget {
                     color: buttonOkColor ?? Colors.green,
                     shape: RoundedRectangleBorder(
                         borderRadius: BorderRadius.circular(buttonRadius)),
-                    onPressed: onOkButtonPressed ?? () {},
+                    onPressed:
+                        onOkButtonPressed, // On passing Null button will not look clickable
```
Comment (xsahil03x) [inline lib/src/asset.dart]: No need to change this part as I already handled the null part by passing an empty method.

Comment (mannprerak2) [reply lib/src/asset.dart]: Passing an empty method will make the button seem clickable but on clicking it user will not see any action, however if you pass null to it, the button will be grayed out and not animate upon clicking.

Comment (xsahil03x) [reply lib/src/asset.dart]: Yes, I got the point. Please fix that other change above this and try resolving the merge conflicts.

**Comment by mannprerak2** [comment]
#22 has solved this, so closing this PR..

**Pull request closed without merging**